include/pybind11/pybind11.h

```diff
@@ -52,6 +52,37 @@ PYBIND11_WARNING_DISABLE_MSVC(4127)
 
 PYBIND11_NAMESPACE_BEGIN(detail)
 
+inline std::string replace_newlines_and_squash(const char *text) {
+    const char *whitespaces = " \t\n\r\f\v";
+    std::string result;
+    bool previous_is_whitespace = false;
+
+    // Replace characters in whitespaces array with spaces and squash consecutive spaces
+    while (*text != '\0') {
+        if (strchr(whitespaces, *text)) {
```

Comment:
Could you please make this `std::strchr`?

Elsewhere we have `std::strcmp` consistently, and we already have `#include <cstring>`.